src/api/scenarioData/scenarioData.service.ts

```
@@ -235,4 +255,283 @@ export class ScenarioDataService {
+          Object.keys(formattedExpectedResultsObject).length > 0
+            ? isEqual(decisionResult.result, formattedExpectedResultsObject)
+            : true;
+
+        const scenarioResult = {
+          inputs: mapTraces(decisionResult.trace, ruleSchema, 'input'),
+          outputs: mapTraces(decisionResult.trace, ruleSchema, 'output'),
+          expectedResults: formattedExpectedResultsObject || {},
+          result: decisionResult.result || {},
+          resultMatch: resultMatches,
+        };
+
+        results[title] = scenarioResult;
+      } catch (error) {
+        console.error(`Error running decision for scenario ${title}: ${error.message}`);
+        results[title] = { error: error.message };
+      }
+    }
+    return results;
+  }
+
+  async generateTestCSVScenarios(
+    filepath: string,
+    ruleContent: RuleContent,
+    simulationContext: RuleRunResults,
+    testScenarioCount?: number,
+  ) {
+    const ruleRunResults: RuleRunResults = await this.generateTestScenarios(
+      filepath,
+      ruleContent,
+      simulationContext,
+      testScenarioCount,
+    );
+
+    const keys = {
+      inputs: extractUniqueKeys(ruleRunResults, 'inputs'),
+      expectedResults: extractUniqueKeys(ruleRunResults, 'expectedResults'),
+      result: extractUniqueKeys(ruleRunResults, 'result'),
+    };
+
+    const headers = [
+      'Scenario',
+      'Results Match Expected (Pass/Fail)',
+      ...this.prefixKeys(keys.inputs, 'Input'),
+      ...this.prefixKeys(keys.expectedResults, 'Expected Result'),
+      ...this.prefixKeys(keys.result, 'Result'),
+    ];
+
+    const rows = Object.entries(ruleRunResults).map(([scenarioName, data]) => [
+      this.escapeCSVField(scenarioName),
+      `n/a`,
+      ...this.mapFields(data.inputs, keys.inputs),
+      ...this.mapFields(data.expectedResults, keys.expectedResults),
+      ...this.mapFields(data.result, keys.result),
+    ]);
+
+    return [headers, ...rows].map((row) => row.join(',')).join('\n');
```

Can we wrap this whole thing in a try/catch, make sure we're throwing errors in the above functions it uses, and then throw an error here too. This way we could return the proper error to the user in `scenarioData.controller` if we wanted.